### ISSUE TITLE
action-setup V2 issue: ERR_PNPM_META_FETCH_FAIL

### ISSUE THREAD

**Issue by ImPrankster** [pnpm/action-setup#136]
Recently when running pnpm/action-setup@v2 in GitHub CI I got this error:
```
GET https://registry.npmjs.org/pnpm: Value of "this" must be of type URLSearchParams
```
Using node-version 20
It happens during self-install, seems to be a connection problem
It started about one day ago

**Comment by alokVishu** [comment]
I have also faced the same issue.

![image](https://github.com/pnpm/action-setup/assets/80033712/cd5c5c75-41db-4eef-ae09-aa38ebe42517)

**Comment by BenjaminNolan** [comment]
Can confirm this started some time between now and 17:00 UTC yesterday, when it was working fine. I suspect NPM have put out an update which broke it, perhaps relating to the deprecation warning for registry.npmjs.org:

```
» curl -I https://registry.npmjs.org     
HTTP/2 200 
date: Thu, 04 Jul 2024 06:13:03 GMT
content-type: application/json
cf-ray: 89dcf7980d501bc4-MAD
cf-cache-status: DYNAMIC
npm-notice: This endpoint is deprecated. Use https://replicate.npmjs.com instead.
server: cloudflare
```

**Comment by alokVishu** [comment]
updated  pnpm/action-setup@v2 =>  pnpm/action-setup@v4 fix the issue

**Comment by dvins** [comment]
Duplicate of #135.

The proposed "fix" is a work around and method of going forward provided GitHub workflows can be readily updated and approved. But it does not resolve stuck / failing workflow runs, many of which cannot be triggered again without additional code updates via merge or rebase of each one to pull in the use of the updated workflow action version.

**Comment by zkochan** [comment]
Fixed in 2.4.1 https://github.com/pnpm/action-setup/releases/tag/v2.4.1
